shell/platform/darwin/ios/framework/Source/FlutterOverlayView.mm

```diff
@@ -21,18 +21,6 @@
 // TODO(amirh): once GL support is in evaluate if we can merge this with FlutterView.
 @implementation FlutterOverlayView
 
-- (instancetype)initWithFrame:(CGRect)frame {
```

Comment:
`google-objc-avoid-throwing-exception`
I ignored this check in `.clang-tidy` since there are a few more tricky ones to fix.  However, these throwing implementations should be removed, they are already marked as `NS_UNAVAILABLE` in the header.
https://github.com/flutter/engine/blob/d5d7526b117dc76ac1cf851ca370be9284ece351/shell/platform/darwin/ios/framework/Source/FlutterOverlayView.h#L33-L34

Reply:
NS_UNAVAILABLE won't catch all usage of these selectors.  How about refactoring this to NSAssert calls?

Reply:
What situations does it not cover?  Are you worried about the selector being send to it?

```objc
@interface Foo : NSObject
- (instancetype)init NS_UNAVAILABLE;
- (instancetype)initWithStuff NS_DESIGNATED_INITIALIZER;
- (void)doStuff;
@end
```
```objc
@interface Bar : Foo
@end
```
<img width="636" alt="Screen Shot 2021-11-16 at 1 52 04 PM" src="https://user-images.githubusercontent.com/682784/142071831-6fa20fba-a5db-4417-b692-4f78d4fddc43.png">
<img width="409" alt="Screen Shot 2021-11-16 at 1 50 10 PM" src="https://user-images.githubusercontent.com/682784/142071534-5a2b75fc-3dbf-4041-baab-eeff390be3a2.png">
<img width="426" alt="Screen Shot 2021-11-16 at 1 49 45 PM" src="https://user-images.githubusercontent.com/682784/142071485-6bd08190-58a4-477f-bba2-2a67395ca3ac.png">

Reply:
Yep, exactly.  Xib's for example will execute `initWithCoder:`.  It uses the objc runtime to execute the selector so NS_UNAVAILABLE won't effect it.

Reply:
You're right, good catch.